pulser-pasqal/pulser_pasqal/pasqal_cloud.py

```diff
@@ -189,7 +191,16 @@ def _fetch_result(self, submission_id: str) -> tuple[Result, ...]:
         meas_basis = seq_builder.get_measurement_basis()
 
         results = []
-        for job in batch.ordered_jobs:
+        sdk_jobs = batch.ordered_jobs
+        if job_ids is not None:
+            ind_job_pairs = [
```

Comment:
Just to confirm, this is so we can retain the returned order that they came in? if we associate and order by the index?

Reply:
Yes, whenever `job_ids` are provided we want to return the results in the same order they were given. This is simultaneously ensuring that's the case and selecting only the relevant subset of jobs